planning/obstacle_velocity_planner/src/node.cpp

```diff
@@ -0,0 +1,689 @@
+    const auto & p_back = input.points.at(i + 1);
+    constexpr double epsilon = 1e-3;
+
+    if (next_length <= trajectory_length_sum + epsilon) {
+      const auto p_interpolate =
+        calcLinearPoint(p_front, p_back, next_length - trajectory_length_sum);
+      output.points.push_back(p_interpolate);
+      next_length += step_length;
+      continue;
+    }
+
+    trajectory_length_sum += tier4_autoware_utils::calcDistance2d(p_front, p_back);
+  }
+
+  output.points.push_back(input.points.back());
+
+  return output;
+}
+
+PredictedPath getHighestConfidencePredictedPath(const PredictedObject & predicted_object)
+{
+  const auto reliable_path = std::max_element(
+    predicted_object.kinematics.predicted_paths.begin(),
+    predicted_object.kinematics.predicted_paths.end(),
+    [](const PredictedPath & a, const PredictedPath & b) { return a.confidence < b.confidence; });
+  return *reliable_path;
+}
+
+bool isAngleAlignedWithTrajectory(
+  const Trajectory & traj, const geometry_msgs::msg::Pose & pose, const double threshold_angle)
+{
+  if (traj.points.empty()) {
+    return false;
+  }
+
+  const double obj_yaw = tf2::getYaw(pose.orientation);
+
+  const size_t nearest_idx = tier4_autoware_utils::findNearestIndex(traj.points, pose.position);
+  const double traj_yaw = tf2::getYaw(traj.points.at(nearest_idx).pose.orientation);
+
+  const double diff_yaw = tier4_autoware_utils::normalizeRadian(obj_yaw - traj_yaw);
+
+  const bool is_aligned = std::abs(diff_yaw) <= threshold_angle;
+  return is_aligned;
+}
+}  // namespace
+
+namespace motion_planning
+{
+ObstacleVelocityPlannerNode::ObstacleVelocityPlannerNode(const rclcpp::NodeOptions & node_options)
+: Node("obstacle_velocity_planner", node_options),
+  self_pose_listener_(this),
+  in_objects_ptr_(nullptr),
+  lpf_acc_ptr_(nullptr),
+  vehicle_info_(vehicle_info_util::VehicleInfoUtil(*this).getVehicleInfo())
+{
+  using std::placeholders::_1;
+
+  // subscriber
+  trajectory_sub_ = create_subscription<Trajectory>(
+    "~/input/trajectory", rclcpp::QoS{1},
+    std::bind(&ObstacleVelocityPlannerNode::onTrajectory, this, _1));
+  smoothed_trajectory_sub_ = create_subscription<Trajectory>(
+    "/planning/scenario_planning/trajectory", rclcpp::QoS{1},
+    std::bind(&ObstacleVelocityPlannerNode::onSmoothedTrajectory, this, _1));
+  objects_sub_ = create_subscription<PredictedObjects>(
+    "~/input/objects", rclcpp::QoS{1},
+    std::bind(&ObstacleVelocityPlannerNode::onObjects, this, _1));
+  odom_sub_ = create_subscription<Odometry>(
+    "~/input/odometry", rclcpp::QoS{1},
+    std::bind(&ObstacleVelocityPlannerNode::onOdometry, this, std::placeholders::_1));
+
+  // publisher
+  trajectory_pub_ = create_publisher<Trajectory>("~/output/trajectory", 1);
+  vel_limit_pub_ = create_publisher<VelocityLimit>("~/output/velocity_limit", 1);
+  clear_vel_limit_pub_ =
+    create_publisher<VelocityLimitClearCommand>("~/output/clear_velocity_limit", 1);
```

Comment:
Node:
https://github.com/autowarefoundation/autoware.universe/pull/940

the durability of velocity limit will be `transient_local`

Reply:
https://github.com/autowarefoundation/autoware.universe/pull/570/commits/cae27e51ebbad376a2f0ffb919ac2957a1132c0a
DONE

Reply:
Note: above PR has not been merged yet.

Reply:
above PR was merged